cep-6.md

```diff
@@ -0,0 +1,252 @@
+
+## Package Plan ##
+
+  environment location: /Users/username/opt/conda_x86_64/envs/geopandas
+
+  added / updated specs:
+    - geopandas
+
+... (output truncated)
+
+
+Channel "defaults" has the following notices:
+  [info] -- Tue May 10 11:50:34 2022
+  This is a test message. It is not very long, and could have a link to a longer post:
+  https://example.com/short-link
+  
+```
+
+### How else can our users access this message?
+
+Additionally, because our users may wish to see this message on demand, we will add a new sub-command called `notices`. 
+The following are a couple examples to show exactly how it would function:
+
+**Basic usage:** grabs notices for all current channels:
+
+```
+$ conda notices
+
+Retrieving notices: done
+
+Channel "defaults" has the following notices:
+  [info] -- Tue May 10 11:50:34 2022
+  This is a test message. It is not very long, and could have a link to a longer post:
+  https://example.com/short-link
+
+Channel "conda-forge" has the following notices:
+  [info] -- Tue May 10 11:50:34 2022
+  Here is another message. It could have info about the latest happenings or blog posts from conda-forge:
+  https://conda-forge.org/
+```
+
+**Show a single channel:** grabs notices for a single channel:
+
+```
+$ conda notices -c defaults
+
+Channel "defaults" has the following notices:
+  [info] -- Tue May 10 11:50:34 2022
+  This is a test message. It is not very long, and could have a link to a longer post:
+  https://example.com/short-link
+```
+
+### What file format will this message have, and what will it contain?
+
+The notification message will be in the JSON file format. This will allow us to not only store the message itself but 
+also metadata about the message, including information about how often the client should display the message (more on 
+this in the next section).
+
+Here's an example of the `notices.json` file which will be stored in the root of the channel directory structure.
+
+```json
+{
+  "notices": [
+    {
+      "id": "1cd1d8e5-d96c-42d1-9c29-e8120ad80823",
+      "message": "Here is an example message that will be displayed to users",
+      "level": "info",
+      "created_at": "2022-04-26T11:50:34+00:00",
+      "expiry": 604800,
+    }
+  ]
+}
+```
+
+Detailed overview of the JSON fields:
+
+- **notices** [Array] holds zero or more notices that will be displayed to the client.
+  - **id** [String] unique ID for the message itself. UUIDs are preferred, but there is no required format.
+  - **message** [String] message that gets displayed to users.
+  - **level** [String] one of (info|warning|critical). These will let our users know the category of the message
+    and will also allow the client to apply different formatting rules (e.g. text color).
+  - **created_at** [String] ISO 8601 formatted timestamp showing the creation time of the message.
+  - **expiry** [Number] starting at `created_at`, a number specifying how long in seconds the message is valid for.
```

Comment:
Hi @chenghlee,

I believe this field initially started out as an "interval" field that would display the message every "x" seconds (seconds being the most granular interval). It then evolved into an `expiry` field, and I thought it would be acceptable to keep it as a number in seconds. On the Python side, I can just create this as a `timedelta` object and then easily calculate expiry time. Implementation here: https://github.com/travishathaway/conda/blob/1ab9deb123ae0d1987850345ea22175f344e31fc/conda/notices/cache.py#L55-L59

Now that I think about it though, creating this as a ISO 8601 would further simplify the logic for determining whether this message is expired....

What do you think? Is it worth implementing this as a timestamp field? The underlying code wouldn't be too difficult to update.

Thanks for your input!

Reply:
> Personally, I find having both `created_at` and `expires_at` as ISO-8601 timestamps much easier to deal with. As an API consumer, having the expiry field as an actual timestamp makes it "obvious" what it is, whereas an integer forces me to look into the spec. Further, a timestamp would make it easier to consume the messages with tooling that doesn't have extensive support for datetime math (e.g., `curl url/to/notices.json | jq '.notices|.[]|[.message, .expires_at]' | ...`).

I also think expiry using ISO 8601 would be far preferable for simplicity reasons.  That's my only real qualm at the moment, so I guess that's a conditional yes from me. :)

Reply:
Thanks for the feedback. I have updated the CEP accordingly 👍